src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -0,0 +1,248 @@
+# Copyright 2019 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+cdef class _HandlerCallDetails:
+    def __cinit__(self, str method, tuple invocation_metadata):
+        self.method = method
+        self.invocation_metadata = invocation_metadata
+
+
+class _ServicerContextPlaceHolder(object): pass
+
+
+cdef class CallbackWrapper:
+    cdef CallbackContext context
+    cdef object future
+
+    def __cinit__(self, object future):
+        self.context.functor.functor_run = self.functor_run
+        self.context.waiter = <cpython.PyObject*>(future)
+        self.future = future
+
+    @staticmethod
+    cdef void functor_run(
+            grpc_experimental_completion_queue_functor* functor,
+            int succeed):
+        cdef CallbackContext *context = <CallbackContext *>functor
+        (<object>context.waiter).set_result(None)
```

Comment:
if the future was done - for example, canceled - this will raise an Exception [1], maybe we could do something like

```python
if not (<object>context.waiter).done():
    (<object>context.waiter).set_result(None)
else:
    logging.debug("Future was cancelled ...."
```


[1] https://github.com/python/cpython/blob/master/Lib/asyncio/futures.py#L231